app/src/main/java/Edu/calvin/cs262/teamc/activity_put_pet_adoption.java

```diff
@@ -0,0 +1,38 @@
+package Edu.calvin.cs262.teamc;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+public class activity_put_pet_adoption extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_put_pet_adoption);
+    }
+
+    public void savepet(View view)
+    {
+        EditText NameText = findViewById(R.id.NameEdit);
+        EditText BreedText = findViewById(R.id.BreedText);
+        EditText AgeText = findViewById(R.id.age);
+
+        if( TextUtils.isEmpty(NameText.getText())){
+            NameText.setError( "Name is required!" );
```

Comment:
Resource-ify strings

Reply:
If you use a string once in a code, then you really shouldn't have any need for "resource" ifying it